include/RevRegFile.h

```diff
@@ -214,21 +218,28 @@ class RevRegFile {
   /// GetX: Get the specifed X register cast to a specific integral type
   template<typename T, typename U>
   T GetX(U rs) const {
+    T res;
     if( IsRV32 ){
-      return RevReg(rs) != RevReg::zero ? T(RV32[size_t(rs)]) : 0;
+      res = RevReg(rs) != RevReg::zero ? T(RV32[size_t(rs)]) : 0;
     }else{
-      return RevReg(rs) != RevReg::zero ? T(RV64[size_t(rs)]) : 0;
+      res = RevReg(rs) != RevReg::zero ? T(RV64[size_t(rs)]) : 0;
     }
+    TRACE_REG_READ(rs,res);
+    return res;
   }
 
   /// SetX: Set the specifed X register to a specific value
   template<typename T, typename U>
   void SetX(U rd, T val) {
+    T res;
     if( IsRV32 ){
-      RV32[size_t(rd)] = RevReg(rd) != RevReg::zero ? uint32_t(val) : 0;
+      res = RevReg(rd) != RevReg::zero ? uint32_t(val) : 0;
+      RV32[size_t(rd)] = res;
     }else{
-      RV64[size_t(rd)] = RevReg(rd) != RevReg::zero ? uint64_t(val) : 0;
+      res = RevReg(rd) != RevReg::zero ? uint64_t(val) : 0;
+      RV64[size_t(rd)] = res;
     }
+    TRACE_REG_WRITE(rd, res);
```

Comment:
Should the trace be in the type of the source value (`T`), or should it be in the type of the destination register (i.e., split into the two `if` conditions above)?